components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManager.java

```diff
@@ -143,6 +155,7 @@ public UserManager getUserManager() throws CharonException {
      */
     private void registerEndpointURLs() {
 
+        AbstractResourceManager.setResourceURLBuilder(new IdentityResourceURLBuilder());
```

Comment:
Why should we populate the static property of the abstract class. Cannot this get passed in as a ResourceManager implementation gets initiated ?

Reply:
In multiple places in charon/scim side static methods in AbstractResourceManager class is getting invoked.